addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -3215,11 +3231,53 @@ options.registry.MegaMenuLayout = options.registry.SelectTemplate.extend({
     _computeWidgetState: function (methodName, params) {
         if (methodName === 'selectTemplate') {
             const templateDefiningClass = this.containerEl.querySelector('section')
-                .classList.value.split(' ').filter(cl => cl.startsWith('s_mega_menu'))[0];
+                .classList.value.split(' ').filter(cl => cl.startsWith('s_mega_menu_'))[0];
             return `website.${templateDefiningClass}`;
         }
         return this._super(...arguments);
     },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * Puts back original template.
+     *
+     * @private
+     * @param {OdooEvent} ev
+     */
+    async _onResetTemplate(ev) {
+        const widgetValue = this._computeWidgetState('selectTemplate');
+        const template = await this._rpc({
+            model: 'ir.ui.view',
+            method: 'render_public_asset',
+            args: [widgetValue, {}],
+        });
+        this.containerEl.insertAdjacentHTML('beforeend', template);
+    },
+});
+
+/**
+ * Hides delete button.
+ */
+options.registry.NoDeleteButton = options.Class.extend({
+    forceNoDeleteButton: true,
+});
+
+/**
+ * Resets Mega Menu template on last column deletion.
+ */
+options.registry.MegaMenuResetOnDelete = options.Class.extend({
+    /**
+     * @override
+     */
+    onRemove() {
+        this._super.apply(this, arguments);
+        if (this.$target.siblings('div[class^="col-"]').length === 0) {
+            core.bus.trigger('mega_menu_reset_template');
```

Comment:
Maybe we should do here what is done in `_onResetTemplate` ? It would avoid the `mega_menu_reset_template` event addition

Reply:
There is two problems with this:
- Using events on the bus is indeed ugly
- It does not account for the async part of the template being reset, source of race conditions.

I think this is possible to code using the `notify` method and the related `option_update` Odoo event. You can `trigger_up` that event, asking for the `MegaMenuLayout` option to do something and account for the async part returned.

Reply:
Not sure about what you mean by "_and account for the async part returned_"

Reply:
Tenir compte de la promesse renvoyée, pas avoir une méthode qui lance des trucs async mais qui n'en tient pas compte.

Reply:
So the problem here is that onRemove will terminate before the action triggered ? 
https://github.com/odoo/odoo/pull/76399/files#diff-70f7fe38208aa7fe678f18e329d3c11b70065dee723921352b6005774e8bab53R3266-R3274

Reply:
Yes. Send a `onSuccess` callback with the option_update event so that it can be called once it is handled and onRemove being an async method, you can wait for that onSuccess call properly.